dot/import_test.go

```diff
@@ -0,0 +1,217 @@
+			247, 140, 177, 132, 53, 34, 61, 138, 218, 245, 234, 4, 194, 75, 26, 135, 102, 227, 220, 1, 235, 3, 204,
+			106, 12, 17, 183, 151, 147, 212, 227, 28, 192, 153, 8, 56, 34, 156, 68, 254, 209, 102, 154, 124, 124,
+			121, 225, 230, 208, 169, 99, 116, 214, 73, 103, 40, 6, 157, 30, 247, 57, 226, 144, 73, 122, 14, 59, 114,
+			143, 168, 143, 203, 221, 58, 85, 4, 224, 239, 222, 2, 66, 231, 168, 6, 221, 79, 169, 38, 12},
+	}
+
+	preRuntimeDigestItem := types.NewDigestItem()
+	err := preRuntimeDigestItem.Set(preRuntimeDigest)
+	require.NoError(t, err)
+	digest.Add(preRuntimeDigestItem.Value())
+
+	sealDigest := types.SealDigest{
+		ConsensusEngineID: types.BabeEngineID,
+		Data: []byte{158, 127, 40, 221, 220, 242, 124, 30, 107, 50, 141, 86, 148, 195, 104, 213, 178, 236, 93, 190,
+			14, 65, 42, 225, 201, 143, 136, 213, 59, 228, 216, 80, 47, 172, 87, 31, 63, 25, 201, 202, 175, 40, 26,
+			103, 51, 25, 36, 30, 12, 80, 149, 166, 131, 173, 52, 49, 98, 4, 8, 138, 54, 164, 189, 134},
+	}
+
+	sealDigestItem := types.NewDigestItem()
+	err = sealDigestItem.Set(sealDigest)
+	require.NoError(t, err)
+	digest.Add(sealDigestItem.Value())
+
+	expectedHeader := &types.Header{
+		ParentHash:     common.MustHexToHash("0x3b45c9c22dcece75a30acc9c2968cb311e6b0557350f83b430f47559db786975"),
+		Number:         big.NewInt(1482002),
+		StateRoot:      common.MustHexToHash("0x09f9ca28df0560c2291aa16b56e15e07d1e1927088f51356d522722aa90ca7cb"),
+		ExtrinsicsRoot: common.MustHexToHash("0xda26dc8c1455f8f81cae12e4fc59e23ce961b2c837f6d3f664283af906d344e0"),
+		Digest:         digest,
+	}
+
+	tests := []struct {
+		name     string
+		filename string
+		want     *types.Header
+		err      error
+	}{
+		{
+			name: "no arguments",
+			err:  errors.New("read .: is a directory"),
+		},
+		{
+			name:     "working example",
+			filename: setupHeaderFile(t),
+			want:     expectedHeader,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newHeaderFromFile(tt.filename)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
+				assert.Equal(t, tt.want, got)
+			}
+		})
+	}
+}
+
+func Test_newTrieFromPairs(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		filename string
+		want     common.Hash
+		err      error
+	}{
+		{
+			name: "no arguments",
+			err:  errors.New("read .: is a directory"),
+			want: common.Hash{},
+		},
+		{
+			name:     "working example",
+			filename: setupStateFile(t),
+			want:     common.MustHexToHash("0x09f9ca28df0560c2291aa16b56e15e07d1e1927088f51356d522722aa90ca7cb"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newTrieFromPairs(tt.filename)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if !tt.want.IsEmpty() {
+				assert.Equal(t, tt.want, got.MustHash())
+			}
```

Comment:
Should we assert got is empty if tt.want is empty?

Reply:
Added check of got for that case.